src/equations/compressible_euler_2d.jl

```diff
@@ -308,7 +308,7 @@ Should be used together with [`UnstructuredMesh2D`](@ref).
   # [DOI: 10.1007/b79761](https://doi.org/10.1007/b79761)
   if v_normal <= 0.0
     sound_speed = sqrt(equations.gamma * p_local / rho_local) # local sound speed
-    p_star = p_local * (1.0 + 0.5 * (equations.gamma - 1) * v_normal / sound_speed)^(2.0 * equations.gamma * equations.inv_gamma_minus_one)
+    p_star = p_local * (1.0 + 0.5 * (equations.gamma - 1) * v_normal / sound_speed)^(2 * equations.gamma * equations.inv_gamma_minus_one)
```

Comment:
I just didn't see it back then :sweat_smile:

Reply:
This was something that Gregor and I discussed in the initial implementation. When to use literals like `1.0` versus `1` because in Fortran these are different. As a follow-up, is it okay to use `1` here because it is adding to other reals but in the if statement we need to use `v_normal <= 0.0` because the `0.0` stand alone?

Reply:
In this case, both versions are equivalent - as long as we use `Float64` for our computations:
```julia
julia> foo(x) = 1 + x
foo (generic function with 1 method)

julia> bar(x) = 1.0 + x
bar (generic function with 1 method)

julia> @code_native syntax=:intel debuginfo=:none foo(1.0)
        .text
        .file   "foo"
        .section        .rodata.cst8,"aM",@progbits,8
        .p2align        3                               # -- Begin function julia_foo_175
.LCPI0_0:
        .quad   0x3ff0000000000000              # double 1
        .text
        .globl  julia_foo_175
        .p2align        4, 0x90
        .type   julia_foo_175,@function
julia_foo_175:                          # @julia_foo_175
        .cfi_startproc
# %bb.0:                                # %top
        movabs  rax, offset .LCPI0_0
        vaddsd  xmm0, xmm0, qword ptr [rax]
        ret
.Lfunc_end0:
        .size   julia_foo_175, .Lfunc_end0-julia_foo_175
        .cfi_endproc
                                        # -- End function
        .section        ".note.GNU-stack","",@progbits

julia> @code_native syntax=:intel debuginfo=:none bar(1.0)
        .text
        .file   "bar"
        .section        .rodata.cst8,"aM",@progbits,8
        .p2align        3                               # -- Begin function julia_bar_173
.LCPI0_0:
        .quad   0x3ff0000000000000              # double 1
        .text
        .globl  julia_bar_173
        .p2align        4, 0x90
        .type   julia_bar_173,@function
julia_bar_173:                          # @julia_bar_173
        .cfi_startproc
# %bb.0:                                # %top
        movabs  rax, offset .LCPI0_0
        vaddsd  xmm0, xmm0, qword ptr [rax]
        ret
.Lfunc_end0:
        .size   julia_bar_173, .Lfunc_end0-julia_bar_173
        .cfi_endproc
                                        # -- End function
        .section        ".note.GNU-stack","",@progbits
```
But `Float64` coefficients stand in the way of running Trixi.jl with `Float32`, so I remove them when I see them from time to time. See also https://docs.julialang.org/en/v1/manual/style-guide/#Avoid-using-floats-for-numeric-literals-in-generic-code-when-possible

Reply:
Aha, gotcha. Thanks for the explanation!